src/fugw/scripts/coarse_to_fine.py

```diff
@@ -264,6 +264,90 @@ def get_neighbourhood_matrix(embeddings, sample, radius):
     return neighbourhood_matrix
 
 
+def compute_sparsity_mask(
+    coarse_mapping,
+    source_sample,
+    target_sample,
+    source_geometry_embeddings,
+    target_geometry_embeddings,
+    source_selection_radius=1,
+    target_selection_radius=1,
+    method="topk",
+):
+    """
+    Compute sparsity mask from coarse mapping.
+
+    Parameters
+    ----------
+    coarse_mapping: fugw.FUGW
+        Dense mapping between subsampled meshes
+    source_sample: torch.Tensor of size (n1)
+        Indices of sampled points on source distribution
+    target_sample: torch.Tensor of size (m1)
+        Indices of sampled points on target distribution
+    source_geometry_embeddings: torch.Tensor of size (n, k)
+        Embeddings approximating the geodesic distance between
+        source vertices
+    target_geometry_embeddings: torch.Tensor of size (m, k)
+        Embeddings approximating the geodesic distance between
+        target vertices
+    source_selection_radius: float
+        Radius used to determine the neighbourhood
+        of source vertices when defining sparsity mask
+    target_selection_radius: float
+        Radius used to determine the neighbourhood
+        of target vertices when defining sparsity mask
+    method: "topk" or "quantile"
+        Method used to select pairs of source and target features
+        whose neighbourhoods will be used to define
+        the sparsity mask of the solution
+
+    Returns
+    -------
+    sparsity_mask: torch.sparse_coo_tensor of size (n, m)
+        Sparsity mask used to initialize the fine mapping.
+    """
+    if method == "quantile":
+        # Method 1: keep first percentile
+        quantile = 99.95
+
+        threshold = np.percentile(coarse_mapping.pi, quantile)
+        rows, cols = np.nonzero(coarse_mapping.pi > threshold)
+
+    elif method == "topk":
```

Comment:
It's actually top1 IIUC ? Maybe a bit too sparse ?

Reply:
Indeed, this implementation only returns the top1 vertex with respect to mass. @alexisthual Any particular reason behind this design choice ?

Reply:
I think my initial guess was that top ranked vertices would usually be in the same area of the cortex, but I don't think it would hurt to try to add more top-k vertices :)

Reply:
I think I will do that in another PR 👍